module/workflow/src/Infrastructure/Mapper/Strategy/StatusAttributeMapperStrategy.php

```diff
@@ -50,30 +51,37 @@ public function map(array $values, ?AggregateId $aggregateId = null): ValueInter
         Assert::allRegex(array_keys($values), '/^[a-z]{2}_[A-Z]{2}$/');
         Assert::notNull($aggregateId);
 
+        /** @var AbstractProduct $aggregate */
         $aggregate = $this->manager->load($aggregateId);
         Assert::isInstanceOf($aggregate, AbstractProduct::class);
 
-        foreach ($values as $key => $value) {
+        foreach ($values as $code => $value) {
             if (null !== $value) {
                 Assert::stringNotEmpty($value);
                 Assert::uuid($value);
-
-                /** @var AbstractProduct $aggregate */
-                $this->statusValidation($key, $aggregate, $value);
+                $this->statusValidation(new Language($code), $aggregate, $value);
             }
         }
 
         return new TranslatableStringValue(new TranslatableString($values));
     }
 
-    private function statusValidation(string $key, AbstractProduct $aggregate, string $value): void
+    private function statusValidation(Language $language, AbstractProduct $aggregate, string $value): void
     {
-        $language = new Language($key);
+        $attributeCode = new AttributeCode(StatusSystemAttribute::CODE);
         $workflow = $this->workflowProvider->provide($language);
-        $statusIds = $this->query->getAvailableStatuses($aggregate, $workflow, $language);
+
+        if ($aggregate->hasAttribute($attributeCode)
+            && $aggregate->getAttribute($attributeCode)->hasTranslation($language)) {
+            $statusIds = $this->query->getAvailableStatuses($aggregate, $workflow, $language);
+        } else {
+            $statusIds = [$workflow->getDefaultStatus()->getValue()];
+        }
 
         if (!in_array($value, $statusIds, true)) {
-            throw new \InvalidArgumentException('This status can\'t be set.');
+            throw new \InvalidArgumentException(
+                sprintf('This status can\'t be set to language %s.', $language->getCode())
```

Comment:
```suggestion
                sprintf('Transition from product %2 cannot be made from status %s to status %s in language %s.', $language->getCode())
```
perhaps? The message would be complete then